client/src/main/java/io/pravega/client/auth/PravegaAuthHandler.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.auth;
+
+import java.util.Map;
+
+public interface PravegaAuthHandler {
+
+    enum  PravegaAccessControlEnum {
+        NONE,
+        READ,
+        READ_UPDATE
+    }
+
+    /**
+     * Returns name of the handler.
+     * @return The unique name assigned to the handler.
+     */
+    String getHandlerName();
```

Comment:
what is the purpose of the handler name?

Reply:
Each custom auth handler is registered with a unique name. A client selects its auth handler by setting a grpc header with a name "method". This is done by implementing `PravegaCredentials` interface and passing it to client calls.